library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -710,6 +711,23 @@ public void emitTuples()
   }
 
   /**
+   * Notifies that the directory is being scanned.<br>
+   * Override this method to custom handling. Will be called once
+   */
+  protected void visitDirectory(Path filePath)
```

Comment:
can you please implement this in one of the subclasses / in a subclass written in test and write a unit test around it? There is not way it is tested as of now.